test/tests/typeguards.test.ts

```diff
@@ -0,0 +1,43 @@
+import { expect } from 'chai';
+import { isDocument, isDocumentArray } from '../../src/utils';
+import { UserRefModel } from '../models/userRefs';
+
+export function typeguards() {
+    it ('should guarantee array of document types', async () => {
+        const UserMaster = await UserRefModel.create({
+            name: 'master'
+        });
+        const UserSub = await UserRefModel.create({
+            name: 'sub',
+            master: UserMaster._id
+        });
+        UserMaster.subAccounts.push(UserSub._id);
+        await UserMaster.populate('subAccounts').execPopulate();
+        if (isDocumentArray(UserMaster.subAccounts)) {
+            expect(UserMaster.subAccounts).to.have.lengthOf(1);
+            for (const doc of UserMaster.subAccounts) {
+                expect(doc.name).to.be.equal('sub');
+                expect(doc.name).to.not.be.equal('other');
+            }
+        } else {
+            throw new Error('Error in isDocumentArray');
+        }
+    });
+    
+    it ('should guarantee single document type', async () => {
+        const UserMaster = await UserRefModel.create({
+            name: 'master'
+        });
+        const UserSub = await UserRefModel.create({
+            name: 'sub',
+            master: UserMaster._id
```

Comment:
```suggestion
            master: UserMaster
```
It works without `._id` and is more clean

Reply:
in practice this almost never happens, you have the ref and you want to explicitly populate, it's this way in order to work as example too

to improve the test as example i will add the _id in another line